app/models/document.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+class Document < ActiveRecord::Base
+
+  validate :validate_event_type, :on => [:create, :update]
+  validate :event_type_already_exist, :on => :create
+
+  EVENTS = [['-- Select --', 0], ['Calendar of events', 'calendar_of_event']].freeze
+
+  scope :calendar_of_event, -> { where(event_type: 'calendar_of_event') }
+
+  validates :url, :file_name, :presence => true, uniqueness: true
+
+  validate :google_document, :on => [:create, :update]
+
+  def validate_event_type
+    errors.add(:event_type, 'Please select event_type') if event_type.to_s == "0"
+  end
+
+
+  def event_type_already_exist
+    return unless Document.where(event_type: event_type).present?
+
+    errors.add(:event_type, "#{event_type.titleize} type already created. Please use another type.")
```

Comment:
Please update the 2nd sentence to say "Please use another type, or use the 'edit' link if you are trying to update "#{event_type.titleize}"